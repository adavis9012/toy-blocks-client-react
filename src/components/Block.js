import React from "react";
import MuiPaper from "@material-ui/core/Paper";
import {Typography} from "@material-ui/core";
import PropTypes from "prop-types";
import withStyles from "@material-ui/core/styles/withStyles";

const Block = ({block}) => {

  return (<React.Fragment>
    {
      block.data.map(el => {
        return (
          <Paper elevation={0} key={`block-${el.id}`}>
            <Number>
              {String(el.attributes.index).padStart(3, '0')}
            </Number>
            <Typography>
              {el.attributes.data}
            </Typography>
          </Paper>
        );
      })
    }
  </React.Fragment>);
}

const Paper = withStyles({
  root: {
    backgroundColor: '#e0e0e0',
    borderRadius: 2,
    marginBottom: 5,
    padding: 5,
  }
})(MuiPaper);

const Number = withStyles({
  root: {
    color: "blue",
    fontSize: "0.8rem",
  }
})(Typography);

Block.propTypes = {
  block: PropTypes.shape({
    data: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        attributes: PropTypes.shape({
          index: PropTypes.number,
          timestamp: PropTypes.number,
          data: PropTypes.string,
          "previous-hash": PropTypes.string,
          hash: PropTypes.string,
        }),
        loading: PropTypes.bool,
        online: PropTypes.bool,
        url: PropTypes.string,
      }).isRequired
    )
  })
};

export default Block;
