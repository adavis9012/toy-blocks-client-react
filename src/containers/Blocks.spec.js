import React from "react";
import { mount } from "enzyme";
import configureMockStore from "redux-mock-store";
import thunk from 'redux-thunk';
import { Provider } from "react-redux";
import { create } from "react-test-renderer";
import ConnectedBlocks, { Blocks } from "./Blocks";
import Block from "../components/Block";

describe("<Blocks />", () => { const actions = {
    checkBlockStatuses: jest.fn()
  };

  const blocks = {
    list: [
      {
        online: false,
        url: 'https://thawing-springs-53971.herokuapp.com',
        data: [
          {
            id: "1",
            attributes: {
              index: 1,
              data: "Blocks 1"
            }
          }
        ],
        loading: false
      },
      {
        online: false,
        url: 'https://secret-lowlands-62331.herokuapp.com',
        data: [
          {
            id: "2",
            attributes: {
              index: 1,
              data: "Blocks 2"
            }
          }
        ],
        loading: false
      }
    ]
  };

  it("should contain <Block />", () => {
    const wrapper = mount(
      <Blocks
        url="https://thawing-springs-53971.herokuapp.com"
        actions={actions}
        blocks={blocks}
      />
    );

    expect(wrapper.find(Block).length).toEqual(1);
  });

  it("should match snapshot", () => {
    const middlewares = [thunk];
    const store = configureMockStore(middlewares)({blocks});
    const component = create(
      <Provider store={store}>
        <ConnectedBlocks
          url="https://thawing-springs-53971.herokuapp.com"
          actions={actions}
          blocks={blocks}
        />
      </Provider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
