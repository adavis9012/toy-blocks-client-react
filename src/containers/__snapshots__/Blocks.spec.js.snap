// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blocks /> should match snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-63"
>
  <div
    className="MuiPaper-root WithStyles(ForwardRef(Paper))-root-3 MuiPaper-elevation0 MuiPaper-rounded"
  >
    <p
      className="MuiTypography-root WithStyles(ForwardRef(Typography))-root-32 MuiTypography-body1"
    >
      001
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Blocks 1
    </p>
  </div>
</div>
`;
